code/modules/ruins/spaceruin_code/hilbertshotel.dm

```diff
@@ -83,9 +114,48 @@ GLOBAL_VAR_INIT(hhmysteryRoomNumber, 1337)
 	checked_in_ckeys |= user.ckey		//if anything below runtimes, guess you're outta luck!
 	if(tryActiveRoom(chosenRoomNumber, user))
 		return
-	if(tryStoredRoom(chosenRoomNumber, user))
-		return
-	sendToNewRoom(chosenRoomNumber, user)
+	//SPLURT EDIT: Commented out until a way to save turfs is found
+	// if(tryStoredRoom(chosenRoomNumber, user, chosen_room))
+	// 	return
+	sendToNewRoom(chosenRoomNumber, user, chosen_room)
+
+/obj/item/hilbertshotel/proc/get_room_type(chosen_room) //Currently unused
+	switch(chosen_room)
+		if("Apartment") return "Apartment"
+		if("Apartment-1") return "Apartment-1"
+		if("Apartment-2") return "Apartment-2"
+		if("Apartment-3") return "Apartment-3"
+		if("Apartment-Bar") return "Apartment-Bar"
+		if("Apartment-Garden") return "Apartment-Garden"
+		if("Apartment-Syndicate") return "Apartment-Syndicate"
+		if("Apartment-Sauna") return "Apartment-Sauna"
+	return "Hotel Room"
+
+/area/hilbertshotel/proc/storeRoom()
+	var/roomSize = (reservation.top_right_coords[1]-reservation.bottom_left_coords[1]+1)*(reservation.top_right_coords[2]-reservation.bottom_left_coords[2]+1)
+	var/storage[roomSize]
+	var/turfNumber = 1
+	var/obj/item/abstracthotelstorage/storageObj = new(storageTurf)
+	storageObj.roomNumber = roomnumber
+	storageObj.parentSphere = parentSphere
+	storageObj.name = "Room [roomnumber] Storage"
+	/* SPLURT EDIT: Commented out the storing function of atoms, 
+	as it's more complex to load them with custom rooms, 
+	due to turfs not being able to be stored.
+	for(var/i=0, i<parentSphere.hotelRoomTemp.width, i++)
+		for(var/j=0, j<parentSphere.hotelRoomTemp.height, j++)
+			var/list/turfContents = list()
+			for(var/atom/movable/A in locate(reservation.bottom_left_coords[1] + i, reservation.bottom_left_coords[2] + j, reservation.bottom_left_coords[3]))
+				if(ismob(A) && !isliving(A))
+					continue //Don't want to store ghosts
+				turfContents += A
+				A.forceMove(storageObj)
+			storage[turfNumber] = turfContents
+			turfNumber++
+	*/
```

Comment:
What if instead of this you make it so that once a room is generated it can only load that exact same room again with the same objects? because I believe some people will wanna keep their objects in a hotel room and come for them later